Source/WebKit/Platform/IPC/unix/ConnectionUnix.cpp

```diff
@@ -192,11 +221,29 @@ bool Connection::processMessage()
     Vector<Attachment> attachments(attachmentCount);
     RefPtr<WebCore::SharedMemory> oolMessageBody;
 
+#if OS(ANDROID)
+    size_t fdIndex = 0;
+    size_t hardwareBufferIndex = 0;
+    for (size_t i = 0; i < attachmentCount; ++i) {
+        switch (attachmentInfo[i].type()) {
+        case AttachmentInfo::Type::FileDescriptor:
+            attachments[attachmentCount - i - 1] = UnixFileDescriptor {
+                attachmentInfo[i].isNull() ? -1 : m_fileDescriptors[fdIndex++],
```

Comment:
Hummm... but in `Connection::sendOutputMessage` we're setting the attachment info as null unconditionally for file descriptors, so wouldn't this always be `-1`?